src/components/DetailContainer/DetailContainer.tsx

```diff
@@ -1,77 +1,29 @@
-import React, { Component, ReactElement, Fragment } from 'react';
+import React, { useState, ChangeEvent, useContext } from 'react';
+import Box from '@material-ui/core/Box';
 
-import { DetailContextConsumer, VersionPageConsumerProps } from '../../pages/Version';
-import Readme from '../Readme';
-import Versions from '../Versions';
-import { preventXSS } from '../../utils/sec-utils';
-import Tabs from '@material-ui/core/Tabs';
-import Tab from '@material-ui/core/Tab';
-import { Content } from './styles';
-import Dependencies from '../Dependencies';
-import UpLinks from '../UpLinks';
+import { DetailContext } from '../../pages/Version';
 
-interface DetailContainerState {
-  tabPosition: number;
-}
+import DetailContainerTabs from './DetailContainerTabs';
+import DetailContainerContent from './DetailContainerContent';
+import { TabPosition } from './tabs';
 
-export const README_LABEL = 'Readme';
-export const DEPS_LABEL = 'Dependencies';
-export const VERSION_LABEL = 'Versions';
-export const UPLINKS_LABEL = 'Uplinks';
+const DetailContainer: React.FC = () => {
+  const [tabPosition, setTabPosition] = useState(TabPosition.README);
+  const detailContext = useContext(DetailContext);
+  const { readMe } = detailContext;
 
-class DetailContainer<P> extends Component<P, DetailContainerState> {
-  public state = {
-    tabPosition: 0,
+  const handleChangeTabPosition = (event: ChangeEvent<{}>) => {
```

Comment:
Here we can use `useCallback`, so the handler is not recreated in each render.
```
  const handleChangeTabPosition = useCallback(
    (event: ChangeEvent<{}>) => {
      const eventTarget = event.target as HTMLSpanElement;
      const chosentab = eventTarget.innerText as TabPosition;
      setTabPosition(TabPosition[chosentab]);
    },
    [setTabPosition]
  );
```

Reply:
https://reactjs.org/docs/hooks-reference.html#usecallback

Reply:
It seems the `event.preventDefault();` was removed.

Reply:
Oh it's true...I have updated the code, now it uses useCallback and event. preventDefault( thank you!!